exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,514 @@
+	}
+	if mark != "" {
+		params.Set("mark", mark)
+	}
+	path := fmt.Sprintf("%s/v%s/%s", lbankAPIURL, lbankAPIVersion, lbankWithdraw)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// RevokeWithdraw cancels the withdrawal given the withdrawalID
+func (l *Lbank) RevokeWithdraw(withdrawID string) (RevokeWithdrawResponse, error) {
+	var resp RevokeWithdrawResponse
+	params := url.Values{}
+	if withdrawID != "" {
+		params.Set("withdrawId", withdrawID)
+	}
+	path := fmt.Sprintf("%s/v%s/%s?", lbankAPIURL, lbankAPIVersion, lbankRevokeWithdraw)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// GetWithdrawalRecords gets withdrawal records
+func (l *Lbank) GetWithdrawalRecords(assetCode, status, pageNo, pageSize string) (WithdrawalResponse, error) {
+	var resp WithdrawalResponse
+	params := url.Values{}
+	params.Set("assetCode", assetCode)
+	params.Set("status", status)
+	params.Set("pageNo", pageNo)
+	params.Set("pageSize", pageSize)
+	path := fmt.Sprintf("%s/v%s/%s", l.APIUrl, lbankAPIVersion, lbankWithdrawalRecords)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// ErrorCapture captures errors
+func ErrorCapture(intermediary json.RawMessage) error {
+	var capErr ErrCapture
+	err := json.Unmarshal(intermediary, &capErr)
+	if err == nil && capErr.Error != 0 {
+		msg, ok := errorCodes[capErr.Error]
+		if !ok {
+			return errors.New("undefined code please check api docs for error code definition")
+		}
+		return errors.New(msg)
+	}
+	return nil
+}
+
+// SendHTTPRequest sends an unauthenticated HTTP request
+func (l *Lbank) SendHTTPRequest(path string, result interface{}) error {
+	var intermediary json.RawMessage
+	err := l.SendPayload(http.MethodGet, path, nil, nil, &intermediary, false, false, l.Verbose, l.HTTPDebugging)
+	if err != nil {
+		return err
+	}
+	err = ErrorCapture(intermediary)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(intermediary, result)
+}
+
+func (l *Lbank) loadPrivKey() error {
+	if l.privKeyLoaded {
+		return nil
+	}
+
+	key := strings.Join([]string{
+		"-----BEGIN RSA PRIVATE KEY-----",
+		l.APISecret,
+		"-----END RSA PRIVATE KEY-----",
+	}, "\n")
+
+	block, _ := pem.Decode([]byte(key))
+	if block == nil {
+		return fmt.Errorf("pem block is nil")
+	}
+
+	p, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("unable to decode priv key: %s", err)
+	}
+
+	var ok bool
+	l.privateKey, ok = p.(*rsa.PrivateKey)
+	if !ok {
+		return fmt.Errorf("unable to parse RSA private key")
+	}
+	l.privKeyLoaded = true
+	return nil
+}
+
+func (l *Lbank) sign(data string) (string, error) {
+	if l.privateKey == nil {
+		return "", errors.New("private key not loaded")
+	}
+	md5hash := common.GetMD5([]byte(data))
+	m := common.StringToUpper(common.HexEncodeToString(md5hash))
+	s := common.GetSHA256([]byte(m))
+	r, err := rsa.SignPKCS1v15(rand.Reader, l.privateKey, crypto.SHA256, s)
+	return common.Base64Encode(r), err
```

Comment:
can check this error and return because if it errors out we still common.Base64Encode(0)